tools/tf_env_collect.sh

```diff
@@ -142,35 +159,33 @@ LD_DEBUG=libs ${python_bin_path} -c "import tensorflow"  2>>${OUTPUT_FILE} > /tm
   fi
   
   
-  echo
-  echo '== nvidia-smi ==================================================='
+  print_header nvidia-smi
   nvidia-smi 2>&1
   
-  echo
-  echo '== cuda libs  ==================================================='
-} >> ${OUTPUT_FILE}
+  print_header 'cuda libs'
 
-find /usr/local -type f -name 'libcudart*'  2>/dev/null | grep cuda |  grep -v "\\.cache" >> ${OUTPUT_FILE}
-find /usr/local -type f -name 'libudnn*'  2>/dev/null | grep cuda |  grep -v "\\.cache" >> ${OUTPUT_FILE}
+  # Find cudart/cudnn files
+  find /usr -type f -name 'libcud*'  2>/dev/null | grep -E 'cuda.*(cudart|cudnn)' | grep -v -F '.cache'
 
-{
-  echo
-  echo '== tensorflow installed from info =================='
-  pip show tensorflow
+  print_header 'tensorflow installation'
+  if ! pip show tensorflow; then
+    echo "tensorflow not found"
+  fi
 
-  echo
-  echo '== python version  =============================================='
+  print_header 'tf_nightly installation'
+  if ! pip show tf_nightly; then
+    echo "tf_nightly not found"
+  fi
+
+  print_header 'python version'
   echo '(major, minor, micro, releaselevel, serial)'
-  python -c 'import sys; print(sys.version_info[:])'
+  "${PYTHON_BIN_PATH}" -c 'import sys; print(sys.version_info[:])'
   
-  echo
-  echo '== bazel version  ==============================================='
+  print_header 'bazel version'
   bazel version
-} >> ${OUTPUT_FILE}
 
-# Remove any words with google.
-mv $OUTPUT_FILE old-$OUTPUT_FILE
```

Comment:
I think they didn't know that you could pipe the output of a subshell through grep, so they did this little dance.